src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -60,17 +49,41 @@
    * <p>Default: 300 seconds
    */
   private int networkTimeoutInMilli = 0; // in milliseconds
-
   /* this should be set to Connection.TRANSACTION_READ_COMMITTED
    * There may not be many implications here since the call to
    * setTransactionIsolation doesn't do anything.
    */
   private int transactionIsolation = Connection.TRANSACTION_NONE;
+  private BaseSFSession sfSession;
+  /** The SnowflakeConnectionImpl that provides the underlying physical-layer implementation */
+  private ConnectionHandler connectionHandler;
 
-  private SFSession sfSession;
+  private boolean showStatementParameters;
 
-  /** Refer to all created and open statements from this connection */
-  private final Set<Statement> openStatements = ConcurrentHashMap.newKeySet();
+  /**
+   * Temporary boolean to detect whether we are using the DefaultConnectionHandler while different
+   * interfaces are in the process of being merged in.
+   */
+  private boolean isUsingDefaultConnectionHandler;
```

Comment:
I don't understand why we need this boolean ? If a consumer of this API initializes the `SnowflakeConnectionV1` class with another implementation of ConnectionHandler, they expect the behavior of that new ConnectionHandler.
The other implementations of the  ConnectionHandler can throw `NotSupportedException` till they implement it.

Reply:
I am also slightly confused by this

Reply:
See the comment below; this is a temporary flag to ensure that unmerged parts of the ConnectionHandler interface won't be called with the wrong handler.